internals/secrethub/run_test.go

```diff
@@ -454,6 +457,48 @@ func TestRunCommand_Run(t *testing.T) {
 				command: []string{"echo", "test"},
 			},
 		},
+		"missing secret": {
+			command: RunCommand{
+				command: []string{"echo", "test"},
+				envar: map[string]string{
+					"missing": "path/to/unexisting/secret",
+				},
+				newClient: func() (secrethub.Client, error) {
+					return fakeclient.Client{
+						SecretService: &fakeclient.SecretService{
+							VersionService: &fakeclient.SecretVersionService{
+								WithDataGetter: fakeclient.WithDataGetter{
+									Err: api.ErrSecretNotFound,
+								},
+							},
+						},
+					}, nil
+				},
+				ignoreMissingSecrets: false,
+			},
+			err: api.ErrSecretNotFound,
+		},
+		"missing secret ignored": {
+			command: RunCommand{
+				command: []string{"echo", "test"},
+				envar: map[string]string{
+					"missing": "path/to/unexisting/secret",
+				},
+				newClient: func() (secrethub.Client, error) {
+					return fakeclient.Client{
+						SecretService: &fakeclient.SecretService{
+							VersionService: &fakeclient.SecretVersionService{
+								WithDataGetter: fakeclient.WithDataGetter{
+									Err: api.ErrSecretNotFound,
+								},
+							},
+						},
+					}, nil
+				},
+				ignoreMissingSecrets: true,
```

Comment:
What does this test? Where do we check that the resulting environment contains `MISSING=""`?

Reply:
It tests that no error is returned. It does indeed not test that the environment of the run command contains the missing key with the empty string as value.

One way to unit-test this would be to extract the logic of constructing the environment from the command, so that it can be tested in isolation (no need to inspect the environment of the other goroutine running the command). This should be done for non-empty secrets as well.

The e2e case could for example be tested by injecting a buffer as stdout and print the secrets to that buffer using the `echo` command (and use no-masking to show them).

In general, this command could use some more testing, both unit- and e2e tests.

Reply:
Alright, if you feel it's tested thoroughly enough for now I'm cool with it. We can schedule more extensive testing for later if you want or add it now. Your call.